osf/models/quickfiles.py

```diff
@@ -0,0 +1,74 @@
+import logging
+
+from osf.models.node import AbstractNode, AbstractNodeManager
+from website.exceptions import NodeStateError
+
+
+logger = logging.getLogger(__name__)
+
+
+class QuickFilesManager(AbstractNodeManager):
+
+    def create_for_user(self, user):
+        possessive_title = get_quickfiles_project_title(user)
+
+        quickfiles, created = QuickFiles.objects.get_or_create(
+            title=possessive_title,
+            creator=user
+        )
+
+        if not created:
+            raise NodeStateError('Users may only have one quickfiles project')
+
+        quickfiles.add_addon('osfstorage', auth=None, log=False)
+
+        return quickfiles
+
+    def get_for_user(self, user):
+        return QuickFiles.objects.get(creator=user)
+
+
+class QuickFiles(AbstractNode):
+    __guid_min_length__ = 10
+
+    objects = QuickFilesManager()
+
+    def __init__(self, *args, **kwargs):
+        kwargs['is_public'] = True
+        super(QuickFiles, self).__init__(*args, **kwargs)
+
+    def remove_node(self, auth, date=None):
+        raise NodeStateError('A QuickFiles node may not be deleted.')
+
+    def set_privacy(self, permissions, *args, **kwargs):
+        if permissions == 'private':
+            raise NodeStateError('A QuickFiles node cannot be made private.')
+        return super(QuickFiles, self).set_privacy(permissions, *args, **kwargs)
```

Comment:
I think this method should error regardless of the value of `permissions`. `set_privacy` should never be called on a quickfiles node.